pynsot/commands/cmd_protocol_types.py

```diff
@@ -0,0 +1,340 @@
+DISPLAY_FIELDS = (
+    ('id', 'ID'),
+    ('name', 'Name'),
+    ('description', 'Description'),
+    ('required_attributes', 'Required Attributes'),
+)
+
+# Fields to display when viewing a single record.
+VERBOSE_FIELDS = (
+    ('id', 'ID'),
+    ('name', 'Name'),
+    ('description', 'Description'),
+    ('required_attributes', 'Required Attributes'),
+    ('site', 'Site ID'),
+)
+
+
+# Main group
+@click.group()
+@click.pass_context
+def cli(ctx):
+    """
+    Protocol Type objects.
+
+    An Protocol Type resource can represent a network protocol type (e.g. bgp, is-is, ospf, etc.)
+
+    Protocol Types can have any number of required attributes as defined below.
+    """
+
+
+# Add
+@cli.command()
+@click.option(
+    '-r',
+    '--required_attributes',
+    metavar='REQUIRED ATTRIBUTES',
+    help='A key/value pair attached to this Protocol Type (format: key=value).',
+    multiple=True,
+    callback=callbacks.transform_attributes,
+    # required=True,
+)
+@click.option(
+    '-e',
+    '--description',
+    metavar='DESCRIPTION',
+    type=str,
+    help='The description for this Protocol Type.',
+)
+@click.option(
+    '-n',
+    '--name',
+    metavar='NAME',
+    type=str,
+    help='The name of the Protocol Type.',
+    required=True,
+)
+@click.option(
+    '-I',
+    '--id',
+    metavar='ID',
+    help='Unique ID of the Protocol Type being retrieved.',
+)
+@click.option(
+    '-s',
+    '--site-id',
+    metavar='SITE_ID',
+    type=int,
+    help='Unique ID of the Site this Protocol Type is under.  [required]',
+    callback=callbacks.process_site_id,
+)
+@click.pass_context
+def add(ctx, required_attributes, description, id, name, site_id):
+    """
+    Add a new Protocol Type.
+
+    You must provide a Protocol Type name or ID using the UPDATE option.
+
+    When adding a new Protocol Type, you must provide a value for the -n/--name
+    option.
+
+    Examples: OSPF, BGP, etc.
+
+    You must also provide attributes, you may specify the -a/--attributes
+    option once for each key/value pair.
+
+    You must provide a Site ID using the -s/--site-id option.
+    """
+    data = ctx.params
+
+    # Remove if empty; allow default assignment
+    if description is None:
+        data.pop('description')
+
+    ctx.obj.add(data)
+
+
+# List
+@cli.group(invoke_without_command=True)
+@click.option(
+    '-r',
+    '--required_attributes',
+    metavar='REQUIRED ATTRIBUTES',
+    help='A key/value pair attached to this Protocol Type (format: key=value).',
+    multiple=True,
+    callback=callbacks.transform_attributes,
+)
+@click.option(
+    '-e',
+    '--description',
+    metavar='DESCRIPTION',
+    type=str,
+    help='Filter by Protocol Type matching this description.',
+)
+@click.option(
+    '-g',
+    '--grep',
+    is_flag=True,
+    help='Display list results in a grep-friendly format.',
+    default=False,
+    show_default=True,
+)
+@click.option(
+    '-I',
```

Comment:
Since ID is relevant here, this should be a lowercase `-i` in keeping with the other object types.

Also, this should be a `type=types.NATURAL_KEY` field since `ProtocolType.name` is the natural key for `ProtocolType`.